portal_sale_personal_data_only/__manifest__.py

```diff
@@ -7,7 +7,8 @@
     "author": "Tecnativa, Odoo Community Association (OCA)",
     "website": "https://github.com/OCA/sale-workflow",
     "license": "AGPL-3",
-    "depends": ["sale"],
+    "depends": ["sale", "portal_account_personal_data_only"],
```

Comment:
Is the dependency over `portal_account_personal_data_only` strictly needed?

Reply:
Yes. Due to the split, these rules had to be split in two, one on each of the modules:

https://github.com/OCA/sale-workflow/blob/53cb00f656baf4ce642d683202d7769e49719d0f/portal_sale_personal_data_only/security/security.xml#L19-L34

Without the dependency, usability would be broken. That's also why this module is marked now as auto install.

Reply:
Yes, but then it's not needed technically if I'm not interpreting it badly. I think the `auto_install` is enough, and now yo can put not only `True` / `False`, but the names of the modules to be installed for this one being auto-installed, and you can include there `portal_account_personal_data_only` instead of a hard dependency.

Reply:
I see what you mean. I also thought about that. However I'm not sure that removing that dependency is the best idea.

On one side, this is a migration. By adding the dependency, we avoid the need of migration scripts, as this module will behave just as it was before. This is good IMHO as it avoids surprises.

OTOH, according to [the docs](https://www.odoo.com/documentation/15.0/developer/reference/backend/module.html?highlight=auto_install#manifest), I can't do it as you said:

> **If it is a list, it must contain a subset of the dependencies**. This module will automatically be installed as soon as all the dependencies in the subset are installed. The remaining dependencies will be automatically installed as well. If the list is empty, this module will always be automatically installed regardless of its dependencies and these will be installed as well.

If I use a module in `auto_install` that's not among its dependencies, it's not guaranteed to work AFAICS.

So, if I have to choose among removing the dependency or removing the `auto_install` feature, IMHO `auto_install` is more valuable, as it makes the whole experience more coherent. After all, it makes sense that, if you want to restrict sale orders visibility, you want the same on invoices.

In the worst case, this is how it was working before the split, so this is not a regression. If you want I can add the idea to the roadmap.

Reply:
OK, let's keep it as it is.